Consolidate all facade projects, cleanup project files

mellinoe opened the pull request:
* Move all of the project configurations currently living in the "facade\<lib>.csproj" files into the root library project. This means there will now be only one project file for the source build of each library, regardless of how many configurations it has. The .builds file of each library will only reference that single project file multiple times.

* Do some minor cleanup of project files along the way. Fix up the VS default configuration blocks to be a bit more uniform. Remove redundant defaults (OutputType, PlatformTarget, etc). Fix some condition spacing.

@ericstj @chcosta 

Chris, this is the work I mentioned to you last week that may have some merge conflicts with the new pkgproj files you are going to add (if those reference the `facade\*.csproj` files that I am deleting). I'm fine with waiting til after your changes go through to merge this in, but I would like to get this in relatively soon because I keep needing to rebase against master when the project.json files are updated in bulk (to update the pre-release versions).

mellinoe commented:
@wes

mellinoe commented:
@weshaggard

bartonjs commented on src/Microsoft.Win32.Primitives/src/Microsoft.Win32.Primitives.csproj:
```diff
@@ -24,10 +21,12 @@
     <Compile Include="System\ComponentModel\Win32Exception.cs" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetsWindows)' == 'true' ">
+  <ItemGroup Condition=" '$(TargetsWindows)' == 'true' And '$(TargetGroup)' != 'net46' ">
```
In my pending change for splitting the crypto contracts across net46 and net461 I found that this style of if was a big pain.  I ended up preferring just switching off of the IsPartialFacadeAssembly property, since that meant I didn't have to list every framework every time.

Of course, I haven't yet had that in front of anyone's eyes to get yelled at.  But I'm just sharing :)

bartonjs commented on src/System.Data.SqlClient/src/project.json:
```diff
@@ -1,41 +1,47 @@
+        "System.Threading": "4.0.10",
+        "System.Threading.Tasks": "4.0.10",
+	"System.Threading.Thread": "4.0.0-rc3-23808",
+	"System.Threading.ThreadPool": "4.0.10-rc3-23808",
```
NIt: This file and whitespace do not seem to be friends.

bartonjs commented on src/System.IO.FileSystem.Watcher/src/project.json:
```diff
@@ -1,20 +1,26 @@
+        "System.Threading": "4.0.10",
+        "System.Threading.Overlapped": "4.0.0",
+        "System.Threading.Tasks": "4.0.10",
+    	"System.Threading.Thread": "4.0.0-rc3-23808"
```
Another whitespace nit, but not quite as egregious

bartonjs commented on src/System.IO.FileSystem/src/System.IO.FileSystem.csproj:
```diff
@@ -10,8 +10,10 @@
     <AssemblyVersion>4.0.1.0</AssemblyVersion>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <EnableWinRT Condition="'$(PackageTargetFramework)' == 'netcore50'">true</EnableWinRT>
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'net46'">true</IsPartialFacadeAssembly>
```
I find myself wondering "is this something that really needs to be in ever package? Can't dir.props set IsPartialFacadeAssembly to true ahead of time when TargetGroup is net45, net451, net452, net46, net461, etc?" (TargetGroup.StartsWith('net4')? :smile:).

Would it ever be wrong?  And, if so, is it fewer places that would need to clear it out than here with libraries needing to set it?

bartonjs commented on src/System.Net.Ping/src/project.json:
```diff
@@ -1,12 +1,18 @@
+      "dependencies": {
+    	"System.Diagnostics.Process": "4.1.0-rc3-23808",
+        "System.Net.Primitives": "4.0.10",
+	"System.Net.NameResolution": "4.0.0-rc3-23808",
```
Whitespace

bartonjs commented on src/System.Net.Security/src/win/project.json:
```diff
@@ -1,15 +1,21 @@
+        "System.Net.Primitives": "4.0.10",
+    	"System.Security.Cryptography.X509Certificates": "4.0.0-rc3-23808",
+        "System.Security.Principal": "4.0.0",
+    	"System.Security.Principal.Windows": "4.0.0-rc3-23808",
```
Whitespace :)

bartonjs commented on src/System.Security.Cryptography.Cng/src/facade/System.Security.Cryptography.Cng.csproj:
```diff
@@ -1,27 +0,0 @@
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Release|AnyCPU'" />
-
-  <ItemGroup>
-    <TargetingPackReference Include="mscorlib" Condition="'$(TargetGroup)' != 'netcore50aot'" />
```
Was this line not important?

bartonjs replied:
(Sort of nevermind; I see that mscorlib made it up into the csproj (line 162); I guess I'm just concerned that this talked about netcore and nothing else is)

mellinoe commented:
Agh, it looks like the project.json's auto-merged badly. I had to fix a bunch of them manually after rebasing, but it looks like some of them auto-merged but with some very weird whitespacing. It's because my version added an extra level of indentation to those lines.

ericstj commented:
This is also going to conflict with @dagood's work to get us onto the latest nuget/dotnet-cli.  Can you hold off on this one until his change goes in?

mellinoe commented:
> This is also going to conflict with @dagood's work to get us onto the latest nuget/dotnet-cli. Can you hold off on this one until his change goes in?

That's fine with me.

dagood commented:
Thanks @mellinoe. I checked in my changes to TFS, so they should show up soon.

This does look like it will be a nasty merge though, sorry about that. Let me know if you have any questions about the changes.

mellinoe commented:
@weshaggard , @chcosta I rebased this against master and just got a clean build. Could you take a look at the packaging properties I modified? For example, I had to change a few lines like this because I added a net46 configuration directly into some projects that didn't have it before:

````XML
-    <UsePackageTargetRuntimeDefaults>true</UsePackageTargetRuntimeDefaults>
+    <UsePackageTargetRuntimeDefaults Condition="'$(TargetGroup)' == ''">true</UsePackageTargetRuntimeDefaults>
````

And I had to do the same for a couple other properties, like `PackageTargetRuntime` which was conditioned on `TargetsWindows` (I added an additional check for `TargetGroup == ''`)

I checked the nuspec files and they look correct to me, just wanted to double check that this was the right way to go about it.

chcosta commented:
Your change to "UsePackageTargetRuntimeDefaults" looks ok to me.
